.clang-tidy

```diff
@@ -45,12 +45,10 @@ Checks: >
   -modernize-make-shared,
   -modernize-use-override,
   performance-*,
-  readability-avoid-const-params-in-decls,
```

Comment:
why deleting this?

Reply:
this is just something that isn't followed at all through the codebase, seems like almost every file has a function that takes by const value so adds a lot of noise when looking at clang-tidy lints, and more of a personal preference lint vs. something that may actually detract from performance

Reply:
If we want to add them back, we should enforce them: i.e. treat these warnings as errors.

Reply:
agreed want to condense this list down to more important lints or things that are already followed first, start treating those as errors, and then slowly expand on the list again